components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -391,115 +372,97 @@ private void addEndpointProperties(Connection dbConnection, String actionId,
+                    return null;
+                },
+                    statement -> {
+                        statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_UUID, actionUUID);
+                        statement.setInt(ActionMgtSQLConstants.Column.TENANT_ID, tenantId);
+                });
+
+            Authentication authentication = null;
+            if (actionEndpointProperties.containsKey(ActionMgtConstants.AUTHN_TYPE_ATTRIBUTE)) {
+                Authentication.Type authnType = Authentication.Type.valueOf(
+                        actionEndpointProperties.get(ActionMgtConstants.AUTHN_TYPE_ATTRIBUTE));
+                switch (authnType) {
+                    case BASIC:
+                        authentication = new Authentication.BasicAuthBuilder(
+                                actionEndpointProperties.get(Authentication.Property.USERNAME.getName()),
+                                actionEndpointProperties.get(Authentication.Property.PASSWORD.getName())).build();
+                        break;
+                    case BEARER:
+                        authentication = new Authentication.BearerAuthBuilder(
+                                actionEndpointProperties.get(Authentication.Property.ACCESS_TOKEN.getName())).build();
+                        break;
+                    case API_KEY:
+                        authentication = new Authentication.APIKeyAuthBuilder(
+                                actionEndpointProperties.get(Authentication.Property.HEADER.getName()),
+                                actionEndpointProperties.get(Authentication.Property.VALUE.getName())).build();
+                        break;
+                    case NONE:
+                        authentication = new Authentication.NoneAuthBuilder().build();
+                        break;
+                    default:
+                        break;
                 }
-
-                return new EndpointConfig.EndpointConfigBuilder()
-                        .uri(endpointUri)
-                        .authentication(authentication).build();
+            } else {
+                ActionMgtConstants.ErrorMessages error =
+                        ActionMgtConstants.ErrorMessages.ERROR_NO_AUTHENTICATION_TYPE;
+                throw new ActionMgtServerException(error.getMessage(), error.getDescription(), error.getCode());
             }
-        } catch (SQLException e) {
-            throw ActionManagementUtil.handleServerException(
-                    ActionMgtConstants.ErrorMessages.ERROR_WHILE_RETRIEVING_ACTION_ENDPOINT_PROPERTIES, e);
+
+            return new EndpointConfig.EndpointConfigBuilder()
+                    .uri(actionEndpointProperties.get(ActionMgtConstants.URI_ATTRIBUTE))
+                    .authentication(authentication).build();
+        } catch (ActionMgtServerException | DataAccessException e) {
+            throw new SQLException(
```

Comment:
Better add a comment on the reason to throw the SQL exception as this exception is really not of SQL type

Reply:
Or we need to have a runtime exception and handle that specifically

Reply:
Added a new Runtime exception with https://github.com/wso2/carbon-identity-framework/pull/6097/commits/d89041e658585942ac7478a30c6e57b5c503f09a